aiohttp/web_reqrep.py

```diff
@@ -794,25 +794,23 @@ def __init__(self, *, body=None, status=200,
                 self._content_dict = {'charset': charset}
                 self.body = text.encode(charset)
             else:
-                self.text = text
                 if content_type or charset:
-                    raise ValueError("Passing both Content-Type header and "
-                                     "content_type or charset params "
-                                     "is forbidden")
+                    raise ValueError('passing both Content-Type header and '
+                                     'content_type or charset params '
+                                     'is forbidden')
+                self.text = text
         else:
             if hdrs.CONTENT_TYPE in self.headers:
                 if content_type or charset:
-                    raise ValueError("Passing both Content-Type header and "
-                                     "content_type or charset params "
-                                     "is forbidden")
-            if content_type:
-                self.content_type = content_type
+                    raise ValueError('passing both Content-Type header and '
+                                     'content_type or charset params '
+                                     'is forbidden')
+            if content_type is None:
```

Comment:
I think this code should be in `StreamResponse` ctor.
`Content-Type` should be `application/octet-stream` if not explicitly specified.
It's true for all responses.

See https://tools.ietf.org/html/rfc7231#section-3.1.1.5